### PR TITLE
External Adaptation DB Support

### PR DESCRIPTION
**Pull request by voxparcxls**
https://jira.jpl.nasa.gov/browse/MSGDS-7442

* New adaptation option for performing database tables creation and DB connection to external(separate) database outside of the cws core database
* Added server adaptation plugin for inserting XML resource to use for JDBC connection
    - file: `server_adaptation.xml`

### CODE REVIEW

File: cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java [hunk 1]
```diff
@@ -2020,6 +2060,28 @@ private static void updateFiles() throws IOException {
...
+			cws_adaptation_db_username != null && cws_adaptation_db_password != null) {
+
+			// Fill in the __ADAPTATION_SERVER_RESOURCE__
+			String serverAdaptationContent = getFileContents(
+				Paths.get(config_work_dir + SEP + "tomcat_conf" + SEP + "server_adaptation.xml"));
+			content = content.replace("<Resource name=\"jdbc/cws\"", serverAdaptationContent);
+
+			content = content.replace("__CWS_ADAPT_DB_URL__",          cws_adaptation_db_url);
+			content = content.replace("__CWS_ADAPT_DB_DRIVER__",       cws_adaptation_db_driver);
+			content = content.replace("__CWS_ADAPT_DB_USERNAME__",     cws_adaptation_db_username);
```
Comment (jamesfwood) [inline cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java:2078]: Can you be consistent on your naming convention?  Make these __CWS_ADAPTATION_ also.

File: cws-core/src/main/java/jpl/cws/core/db/DbService.java [hunk 2]
```diff
@@ -9,6 +9,8 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+//import javax.annotation.Resource;
```
Comment (galenatjpl) [inline cws-core/src/main/java/jpl/cws/core/db/DbService.java:13]: me just being picky, but no need to keep lines like this, in the code

File: cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java [hunk 3]
```diff
@@ -1412,6 +1422,34 @@ private static void showInstallationInfo() {
...
+		cws_adaptation_db_username = getPreset("adaptation_db_username");
+		cws_adaptation_db_password = getPreset("adaptation_db_password");
+		cws_adaptation_db_url = "jdbc:" + cws_adaptation_db_type + "://" + cws_adaptation_db_host + ":" + cws_adaptation_db_port + "/" + cws_adaptation_db_name + "?autoReconnect=true";
+
+		if (cws_adaptation_use_shared_db != null) {
+			if (cws_adaptation_use_shared_db.equalsIgnoreCase("y")) {
+
+				if (cws_adaptation_db_type.equals("mariadb")) {
+					cws_adaptation_db_driver = "org.mariadb.jdbc.Driver";
+				} else {
```
Comment (galenatjpl) [inline cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java:1440]: not sure if we want to make this assumption implicitly that mysql is always the alternate, but probably correct.  Better to be explicit in my opinion to more future code more robust.

File: cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java (hunk 1 above)
Comment (galenatjpl) [reply cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java:2078]: I would also vote to be consistent, but maybe have both be the shorter "ADAPT" form?

File: install/tomcat_conf/server_adaptation.xml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+          maximumPoolSize="5"
+          connectionTimeout="300000"
+
+          driverClassName="__CWS_ADAPT_DB_DRIVER__"
+          jdbcUrl="__CWS_ADAPT_DB_URL__"
+          username="__CWS_ADAPT_DB_USERNAME__"
+          password="__CWS_ADAPT_DB_PASSWORD__"
+
+          transactionIsolation="TRANSACTION_READ_COMMITTED" />
```
Comment (galenatjpl) [inline install/tomcat_conf/server_adaptation.xml:15]: This database is for adaptation (non-camunda tables), and therefore does not have to have the TRANSACTION_READ_COMMITTED level.  This level is a requirement of Camunda, but not a usual requirement of other database purposes.  I would probably just leave this line out.

**Review by galenatjpl — CHANGES_REQUESTED** [review]
This looks good overall, but I left a few minor/optional comments, and one (the transaction read committed level) comment that might be more of a mandatory change for this PR.  Thanks!

File: cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java (hunk 1 above)
Comment (voxparcxls) [reply cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java]: renamed the cws adapt variables to `cws_adapt_`

File: cws-core/src/main/java/jpl/cws/core/db/DbService.java (hunk 2 above)
Comment (voxparcxls) [reply cws-core/src/main/java/jpl/cws/core/db/DbService.java]: Removed

File: cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java (hunk 3 above)
Comment (voxparcxls) [reply cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java]: Set "mysql" condition

File: install/tomcat_conf/server_adaptation.xml (hunk 4 above)
Comment (voxparcxls) [reply install/tomcat_conf/server_adaptation.xml]: Removed `TRANSACTION_READ_COMMITTED ` line

**Pull request merged**